src/components/ProjectDetails/ProjectDetails.js

```diff
@@ -152,15 +158,15 @@ const ProjectDetails = ( {
             <Button
               level="neutral"
               text={t( "JOIN" )}
-              onPress={joinProject}
+              onPress={() => setOpenSheet( JOIN )}
               loading={loadingProjectMembership}
```

Comment:
In addition to showing a loading indicator, I think we should disable this button while the join request is in flight.

Reply:
Done